py3status/modules/speedtest.py

```diff
@@ -0,0 +1,273 @@
+        "up [\?color=upload {upload} {upload_unit}] "
+        "down [\?color=download {download} {download_unit}]]"
+    )
+    server_id = None
+    si_units = False
+    thresholds = {
+        "download": [(0, "bad"), (1024, "degraded"), (1024 * 1024, "good")],
+        "ping": [(200, "bad"), (150, "orange"), (100, "degraded"), (10, "good")],
+        "upload": [(0, "bad"), (1024, "degraded"), (1024 * 1024, "good")],
+    }
+    timeout = 10
+    unit_bitrate = "MB/s"
+    unit_size = "MB"
+
+    def post_config_hook(self):
+        if not self.py3.check_commands("speedtest-cli"):
+            raise Exception(STRING_NOT_INSTALLED)
+
+        self.button_refresh = 2
+        self.sleep_timeout = 2.5
+        self.speedtest_data = {}
+        self.cached_until = self.py3.CACHE_FOREVER
+        self.placeholders = self.py3.get_placeholders_list(self.format)
+        self.thresholds_init = self.py3.get_color_names_list(self.format)
+        self.command = "speedtest-cli --json --secure"
+        self.url = None
+
+        # if download* or upload* missing, run complete test
+        if any(
+            key.startswith(x)
+            for x in ["download", "upload"]
+            for key in self.placeholders
+        ):
+            if not any(key.startswith("upload") for key in self.placeholders):
+                self.command += " --no-upload"
+
+            if not any(key.startswith("download") for key in self.placeholders):
+                self.command += " --no-download"
+
+        if self.server_id and int(self.server_id):
+            self.command += " --server " + str(self.server_id)
+
+        if self.button_share and all(
+            x in self.placeholders for x in ["download", "upload"]
+        ):
+            self.command += " --share"
+
+        self.command += " --timeout " + str(self.timeout)
+
+    def _is_running(self):
+        try:
+            self.py3.command_output(["pgrep", "speedtest-cli"])
+            return True
+        except self.py3.CommandError:
+            return False
+
+    def _cast_number(self, value):
+        try:
+            value = int(value)
+        except ValueError:
+            try:
+                value = float(value)
+            except ValueError:
+                pass
+        return value
+
+    def _get_speedtest_data(self):
+        return loads(self.py3.command_output(self.command))
+
+    def _start(self):
+        self.cached_until = self.py3.time_in(self.sleep_timeout)
+
+        if not self._is_running():
+            self.cached_until = self.py3.CACHE_FOREVER
+            previous_data = self.py3.storage_get("speedtest_data") or {}
+            current_data = self._get_speedtest_data() or {}
+
+            if len(current_data) > 1:
+                # create a "total" for know if cnx is better or not
+                # between two run
+                current_data["total"] = int(current_data.get("download", 0)) + int(
+                    current_data.get("upload", 0)
```

Comment:
(Q) What is this for? It's the only `total` words in the code. Is this meant to be removed too (quality)?